user_service/services/email_service.go

```diff
@@ -0,0 +1,51 @@
+package services
+
+import (
+	"fmt"
+	"user_service/config"
+
+	"gopkg.in/gomail.v2"
+)
+
+type IEmailService interface {
+	SendEmail(token string, receiverEmail string) error
+}
+
+type EmailService struct {
+}
+
+func NewEmailService() IEmailService {
+	return &EmailService{}
+}
+
+func (es *EmailService) SendEmail(token string, receiverEmail string) error {
+	verificationLink := fmt.Sprintf("http://localhost:8080/verify-user/%s", token)
+	emailBody := `
+	<html>
+		<body>
+			<p>Hello!</p>
+			<p>Click the following link to verify your email address:</p>
+			<p><a href="` + verificationLink + `">Verify Email Address</a></p>
+		</body>
+	</html>
+	`
+
+	fmt.Println(config.AppConfig.SENDER_EMAIL,
+		config.AppConfig.SMTP_SERVER,
+		config.AppConfig.SMTP_PORT,
+		config.AppConfig.SENDER_EMAIL,
+		config.AppConfig.SENDER_APP_PASS)
+
+	m := gomail.NewMessage()
+	m.SetHeader("From", config.AppConfig.SENDER_EMAIL)
```

Comment:
`config.AppConfig.SENDER_EMAIL` This could be made part of EmailService struct as an attribute

Reply:
I need to make the appI need to make app_pass smtpServer smtpPort part of the struct as well. cuz this is the configuration which may change if email changes  btw I got the point why you want me to make it part of struct